core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -1181,13 +1185,34 @@ bool
 jit_compile_op_return(JitCompContext *cc, uint8 **p_frame_ip)
 {
     JitBlock *block_func = cc->block_stack.block_list_head;
+    JitInsn *insn = NULL;
+    JitBasicBlock *basic_block = NULL;
 
-    bh_assert(block_func);
+    bh_assert(block_func && LABEL_TYPE_FUNCTION == block_func->label_type);
 
-    handle_func_return(cc, block_func);
+    /* end cur_basic_block */
     SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
+    gen_commit_values(cc->jit_frame, cc->jit_frame->lp, cc->jit_frame->sp);
+    clear_values(cc->jit_frame);
+
+    insn = GEN_INSN(JMP, 0);
+    if (!insn) {
+        jit_set_last_error(cc, "generate jmp insn failed");
+        goto fail;
+    }
+    if (!jit_block_add_incoming_insn(block_func, insn)) {
+        jit_set_last_error(cc, "add incoming insn failed");
+        goto fail;
+    }
+
+    /* start a new basic block */
+    CREATE_BASIC_BLOCK(basic_block);
+    SET_BB_BEGIN_BCIP(basic_block, *p_frame_ip);
+    SET_BUILDER_POS(basic_block);
```

Comment:
No need to create new basic block, it has no processor, no other code will jump to it.

Reply:
No need to create the new basic block, the next `handle_next_reachable_block()` will find a new jit block, create it's end or else basic block if it is not created, and then start to translate that basic block.